### PR TITLE
chore: update dependencies, resolve dependency issues.

### PR DESCRIPTION
**Pull request by autonomousapps**
* Updated several plugins:
    * Dependency Analysis Gradle Plugin, which required...
    * Kotlin to 1.9.x, which required...
    * Dokka to 1.9.10, and
    * Vanniktech Maven Publish plugin 0.27.0.
* The Kotlin update also necessitated some code changes. I'm not sure I resolved those in the best manner.

I then ran `buildhealth` (which didn't work before the above changes) and resolved the issues it reported. I punted on the `test` advice to keep the PR small. (Out of curiosity, why do you run `projectHealth` instead?)

There are some ABI changes. From a quick glance, it looks like they're a new method added to enum classes by newer Kotlin.

### CODE REVIEW

File: misk-grpc-reflect/build.gradle.kts [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies {
   implementation(libs.kotlinLogging)
   implementation(libs.kotlinReflect)
   implementation(libs.okio)
-  implementation(libs.wireGrpcClient)
+//  implementation(libs.wireGrpcClient)
```
Comment (jvmakine) [inline misk-grpc-reflect/build.gradle.kts:35]: Should we just remove these?

**Review by jvmakine — APPROVED** [review]

File: misk-grpc-reflect/build.gradle.kts (hunk 1 above)
Comment (autonomousapps) [reply misk-grpc-reflect/build.gradle.kts:35]: yes for sure. I should have done that before converting from draft to ready for review.

Is there any documentation on how to update the ABI baselines?

Comment (jvmakine) [reply misk-grpc-reflect/build.gradle.kts:35]: I think you can run `gradle apiDump` and commit the changed `.api` files

Comment (jvmakine) [reply misk-grpc-reflect/build.gradle.kts:35]: The contributing guide should have more info here: https://github.com/cashapp/misk/blob/master/CONTRIBUTING.md#breaking-changes

**Comment by autonomousapps** [comment]
@jvmakine I don't have authority to merge PRs into this repo, fyi.

**Pull request merged**